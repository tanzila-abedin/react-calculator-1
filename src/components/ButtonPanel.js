import PropTypes from 'prop-types';
import Button from './Button';

const buttonGroups = {
  group_1: ['AC', '+/-', '%', '/'],
  group_2: ['7', '8', '9', 'X'],
  group_3: ['4', '5', '6', '-'],
  group_4: ['1', '2', '3', '+'],
  group_5: ['0', '.', '='],
};

const ButtonPanel = (props) => (
  <div data-testid="button-panel">
    <div>
      {buttonGroups.group_1.map((button) => (
        <Button
          key={button}
          buttonName={button}
          handleClick={props.handleClick}
        />
      ))}
    </div>
    <div>
      {buttonGroups.group_2.map((button) => (
        <Button
          key={button}
          buttonName={button}
          handleClick={props.handleClick}
        />
      ))}
    </div>
    <div>
      {buttonGroups.group_3.map((button) => (
        <Button
          key={button}
          buttonName={button}
          handleClick={props.handleClick}
        />
      ))}
    </div>
    <div>
      {buttonGroups.group_4.map((button) => (
        <Button
          key={button}
          buttonName={button}
          handleClick={props.handleClick}
        />
      ))}
    </div>
    <div>
      {buttonGroups.group_5.map((button) => (
        <Button
          key={button}
          buttonName={button}
          handleClick={props.handleClick}
        />
      ))}
    </div>
  </div>
);

ButtonPanel.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default ButtonPanel;
