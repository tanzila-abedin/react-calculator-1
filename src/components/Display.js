import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="display" data-testid="display">
    {result}
  </div>
);

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
