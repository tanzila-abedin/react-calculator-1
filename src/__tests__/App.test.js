import renderer from 'react-test-renderer';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

it('renders correctly', () => {
  const app = renderer
    .create(
      <div className="App">
        <header>
          <h2>
            Calculator App built using React
          </h2>
          <p>
            Built by Ryel Banfield
          </p>
        </header>
        <div className="calculator">
          <Display />
          <ButtonPanel />
        </div>
      </div>,
    )
    .toJSON();
  expect(app).toMatchSnapshot();
});
