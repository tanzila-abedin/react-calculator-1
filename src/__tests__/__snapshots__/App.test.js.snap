// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <h2>
      Calculator App built using React
    </h2>
    <p>
      Built by Ryel Banfield
    </p>
  </header>
  <div
    className="calculator"
  >
    <div
      className="display"
      data-testid="display"
    >
      0
    </div>
    <div
      data-testid="button-panel"
    >
      <div>
        <button
          className="hvr-ripple-out"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
      </div>
      <div>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="hvr-fade"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="hvr-ripple-out equalButton"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
